Homework 7

Zenixas opened the pull request:
Fully added both console and file loggers. Easy ability to change from one to another inside ActiveLogger.cs file.

Zenixas committed:
Homework 7

Fully added both console and file loggers. Easy ability to change from one to another inside ActiveLogger.cs file.

Zenixas committed:
Added class modifiers in some methods

Added class modifiers in some methods,
Capitalized ILogger.

Almantask commented on RecipeApp/RecipeApp.Core/Exceptions/Exceptions.cs:
```diff
@@ -0,0 +1,21 @@
+    {
+        public InvalidRecipeException(string reason) : base(reason)
+        {
+            ActiveLogger.iLogger.Log(reason, LoggingLevels.levels.Warn);
```
Should be.Instance. ILogger is not pascal case and besides by saying logger twice (in class and field name) you are redundant.

Almantask commented on RecipeApp/RecipeApp.Core/Services/Logging/ILogger.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public interface ILogger
+    {
+        public abstract void Log(string message, LoggingLevels.levels level);
```
Why abstract?

Zenixas replied:
I thought that it needed to be abstract, because I would need to create either console logger or file logger. But I might have misunderstood what it means.

Almantask commented on RecipeApp/RecipeApp.Core/Services/Logging/Types/FileLogging.cs:
```diff
@@ -0,0 +1,34 @@
+        {
+            using (StreamReader reader = new StreamReader(LogsFile))
+            {
+                return reader.ReadToEnd();
```
Streams are usually fro more advanced cases. Use FilemWirteAllText, ReadAllText instead

Almantask commented on RecipeApp/RecipeApp.Core/Services/RecipeConversion/ConvertToSiUnitsClass.cs:
```diff
@@ -0,0 +1,50 @@
+
+namespace RecipeApp.Core
+{
+    internal class ConvertToSiUnitsClass : RecipeConverter
```
Dotn use class in name

Zenixas replied:
Understood, I had method with same exact name, so added word class to distinguish them. But I'll change them

Zenixas committed:
Fixed up the issues, renamed, removed unnecessaries

- Renamed the files and classes containing "Class" in their name.
- Renamed "ActiveLogger" class to "Instance" to avoid redundancies.
- Replaced streams with File.WriteAllText and File.ReadAllText methods
- Removed the unnecessary Abstract keyword.

Zenixas commented:
- Renamed the files and classes containing "Class" in their name.
- Renamed "ActiveLogger" class to "Instance" to avoid redundancies.
- Replaced streams with File.WriteAllText and File.ReadAllText methods
- Removed the unnecessary Abstract keyword.
